util/vim/vim.go

```diff
@@ -0,0 +1,84 @@
+package vim
+
+const (
+	// NormalMode : every key is interpreted as focus movement,
+	// let it be from containers or text lines.
+	NormalMode int = iota	// 0
+	// InsertMode : every key is sent to the program as normal,
+	// such as input boxes.
+	InsertMode				// 1
+	// TODO
+	VisualMode				// 2
+
+	// Default: disabled
+	Disabled = -1
```

Comment:
Why do we need disabled? Isn't this what `VimEnabled: false` in the config is for?
And if I missunderstand, why isn't this `0`? You might also want to create a type alias `type VimMode int`.

Reply:
Config's VimEnabled is only used when instantiating vim.Vim. This struct then holds a value, CurrentMode, that will be checked everywhere vim mode needs to replace normal bindings.